src-rust/toolkit-cli/src/workflow/select.rs

```diff
@@ -1,48 +1,87 @@
     print!("请输入要选题的文章的 URL：");
-    let _= stdout().flush();
+    let _ = stdout().flush();
     let mut url = String::new();
     stdin().read_line(&mut url).unwrap_or(0);
     let url = url.trim();
     println!("您选择的文章 URL 是：{}", url);
 
-    // Get markdown content
-    let markdown = get_content(url);
-
-    match markdown {
-        Ok(markdown) => {
-            println!("已成功转换为 Markdown");
-
-            // TODO: Be able to allow the user to set the file name
-            let file_name = "article.md";
-            std::fs::write(file_name, markdown).expect("无法写入文件");
-            println!("已将 Markdown 内容写入文件：{}", file_name);
-
-            let path_to_file = std::env::current_dir().unwrap().join(file_name);
-            let editor = &config.editor;
-            let _= std::process::Command::new(editor)
-                .arg(path_to_file)
-                .status()
-                .expect("无法打开编辑器");
-            // We would not be able to detect the program's exit status if it is
-            // a program that only uses the terminal to launch the editor.
-            // e.g. Visual Studio Code.
-
-            println!("如果您已经完成了编辑，请输入任何内容以继续。");
-            let _= stdin().read_line(&mut String::new());
-        }
-        Err(e) => {
-            eprintln!("Markdown 转换失败：{}", e);
-        }
+    // Get the type of the article
+    let article_types = &repo_rule.articles;
+    println!("请选择文章类型：");
+    for (i, article_type) in article_types.iter().enumerate() {
+        println!("{}. {}", i + 1, article_type.description);
+    }
+    print!("请输入文章类型的编号：");
+    let _ = stdout().flush();
+    let mut article_type_index = String::new();
+    stdin().read_line(&mut article_type_index).unwrap_or(0);
+    let article_type_index = article_type_index.trim().parse::<usize>().unwrap_or(0);
+    let article_type = &article_types[article_type_index - 1];
+    println!("您选择的文章类型是：{}", article_type.description);
+
+    let mut vars = HashMap::new();
+    vars.insert("via", url);
+    vars.insert("selector", &user);
+
+    let article = select_article(&repo_rule, article_type, &vars).await;
+    if article.is_err() {
+        eprintln!("Failed to select article: {:?}", article.err());
+        return;
     }
+    let article = article.unwrap();
+
+    // TODO: Be able to allow the user to set the file name
+    let file_name = "article.md";
+    std::fs::write(file_name, article).expect("无法写入文件");
+    println!("已将 Markdown 内容写入文件：{}", file_name);
+
+    let path_to_file = std::env::current_dir().unwrap().join(file_name);
+    let editor = &config.editor;
+    let _ = std::process::Command::new(editor)
+        .arg(path_to_file)
+        .status()
+        .expect("无法打开编辑器");
+    // We would not be able to detect the program's exit status if it is
+    // a program that only uses the terminal to launch the editor.
+    // e.g. Visual Studio Code.
+
+    println!("如果您已经完成了编辑，请输入任何内容以继续。");
+    let _ = stdin().read_line(&mut String::new());
```

Comment:
This could be done when toolbox itself is an extension of the editor.
But the current codes also has no problem.

Reply:
I thought that we said there won't be an editor part in the toolkit...

Unless I misunderstood your comment, would you further explain?